core/src/workflows/get_link_result.rs

```diff
@@ -33,42 +26,85 @@ pub async fn get_link_result_workflow<'a>(
     let (link_results, errors): (Vec<_>, Vec<_>) = links
         .iter()
         .map(|link| {
-            //we should probably replace this with get_entry_result_workflow, it does all the work needed
-            context
-                .block_on(
-                    get_entry_with_meta_workflow(&context, &link, &link_args.options.timeout).map(
-                        |link_entry_result| {
-                            link_entry_result
-                                .map(|link_entry_option| {
-                                    link_entry_option.map(|link_entry| {
-                                        let headers = if link_args.options.headers {
-                                            link_entry.headers
-                                        } else {
-                                            Vec::new()
-                                        };
-                                        Ok(LinksResult {
-                                            address: link.clone(),
-                                            headers,
-                                        })
-                                    })
-                                })
-                                .unwrap_or(None)
-                        },
-                    ),
-                )
-                .unwrap_or(Err(HolochainError::ErrorGeneric(
-                    "Could not get links".to_string(),
-                )))
+            get_latest_entry(
+                context.clone(),
+                link.clone(),
+                link_args.options.timeout.clone(),
+            )
+            .map(|link_entry_result_option| {
+                link_entry_result_option.map(|link_entry| {
+                    let headers = if link_args.options.headers {
+                        link_entry.headers
+                    } else {
+                        Vec::new()
+                    };
+                    Ok(LinksResult {
+                        address: link.clone(),
+                        headers,
+                        crud_status: link_entry.entry_with_meta.crud_status,
+                        crud_link: link_entry.entry_with_meta.maybe_link_update_delete,
+                    })
+                })
+            })
+            .unwrap_or(None)
```

Comment:
I don't think the above is logically correct. `get_links` above in line 19 returns the target addresses of all matching links. This means that `get_latest_entry` in line 29 gets the target of the link, which in turn means that the `headers` that are **returned are the headers of the link target** and not the headers of the link (i.e. the LinkAdd entries). Or am I confused?

Reply:
You are absolutely right, GetLinks is returning link targets because it returns this list of link targets this includes CrudStatus and CrudLink. It is essentially doing the same thing as it was doing in the last pull request which gets headers(as per app_spec results). I think there was some miscommunication in which headers we were talking about(I think I even asked this in heatbeat)

-It brings about questions. When we mention crud_stauts of link what are we talking about? target or linkAdd itself.
-CrudLink? Target or itself?
I ask these because for a system type such as linkadd these will mostly never change so there is a question of why should they be returned in the first place. I think there might have been a breakdown of requirements